### PR TITLE
Add formatting workflow

### PR DESCRIPTION
**Pull request by Shane32**
- Update editorconfig to match other repos
- Add workflow to check formatting for each solution
- Run `dotnet format` on all solutions
- A couple misc changes so formatting checks pass

`dotnet format` is done on a per-solution basis, so I quickly just made the test run in a matrix for each solution/csproj.

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Check formatting
+
+on:
+  pull_request:
```
Comment (sungam3r) [inline .github/workflows/format.yml:4]:
```suggestion
  pull_request:
    branches:
      - master
```

Do you think branch specification is not necessary for examples repo?

Comment (Shane32) [reply .github/workflows/format.yml:4]: I would not do branch specification for any repo. It prevents checks when creating a pr that targets another pr

Comment (sungam3r) [reply .github/workflows/format.yml:4]: I thought about branch spec as about good thing that helps to deal with temporary work merging one pr into another, yes. Now I'm 50/50 about its benefit. It seems that cases of PR to anbother PR are quite rare and only we make them. And I agree that in practice I want to see the results of checks.

**Review by sungam3r — APPROVED** [review]
As I understand it's worth to continue keeping all usings explicit here.

**Pull request merged**